app/models/mixins/relationship_mixin.rb

```diff
@@ -390,6 +390,22 @@ def subtree_arranged(*args)
     Relationship.arranged_rels_to_resources(subtree_rels_arranged(*args))
   end
 
+  def grandchild_rels(*args)
+    options = args.extract_options!
+    rels = relationships.inject(Relationship.none) do |stmt, r|
+      stmt.or(Relationship.where(r.grandchild_conditions))
+    end
+    Relationship.filter_by_resource_type(rels, options)
+  end
+
+  def child_and_grandchild_rels(*args)
+    options = args.extract_options!
+    rels = relationships.inject(Relationship.none) do |stmt, r|
+      stmt.or(Relationship.where(r.child_and_grand_conditions))
```

Comment:
tests failing because `child_and_grand_conditions` should be `child_and_grandchild_conditions`